### PR TITLE
fix for , non presence of consul-version meta

### LINKED ISSUE #18406: UI Nodes view is broken in Consul v1.16.1
#### Overview of the Issue
After upgrading from Consul v1.16.0 to Consul v1.16.1 the Nodes view in UI is broken.

---

#### Reproduction Steps

Upgrade Consul from v1.16.0 to Consul v1.16.1 with the official HashiCorp Consul Chart v1.2.0 by using:
`image: "hashicorp/consul:1.16.1"`

### PR DESCRIPTION
**Pull request by vijayraghav-io**
Fixes #18406 

Hi @david-yu 
Fixed the ui for non-presence of consul-version in Node meta.

### CODE REVIEW

File: ui/packages/consul-ui/app/serializers/application.js [hunk 1]
```diff
@@ -227,36 +227,38 @@ export default class ApplicationSerializer extends Serializer {
     // create a Set and add version with only major.minor : ex-1.24.6 as 1.24
     let versionSet = new Set();
     payload.forEach(function (item) {
-      if (item.Meta && item.Meta['consul-version'] !== '') {
+      if (item.Meta && item.Meta['consul-version'] && item.Meta['consul-version'] !== '') {
```
Comment (natmegs) [inline ui/packages/consul-ui/app/serializers/application.js:230]:
```suggestion
      if (item.Meta && item.Meta['consul-version']) {
```

Checking for truthiness here should be sufficient :)

File: ui/packages/consul-ui/app/serializers/application.js [hunk 2]
```diff
@@ -227,36 +227,38 @@ export default class ApplicationSerializer extends Serializer {
...
-      // Split the versions into arrays of numbers
-      const versionA = a.split('.').map((part) => {
-        const number = Number(part);
-        return isNaN(number) ? 0 : number;
-      });
-      const versionB = b.split('.').map((part) => {
-        const number = Number(part);
-        return isNaN(number) ? 0 : number;
-      });
+    if (versionArray.length > 0) {
```
Comment (natmegs) [inline ui/packages/consul-ui/app/serializers/application.js:238]: I think we can remove this as sort will be a noop if the array is empty

**Review by natmegs — COMMENTED** [review]
Hey @vijayraghav-io thanks so much for this contribution!! This is working great and we really appreciate this bug fix

It would be great to add a unit test for this case and the `getDistinctConsulVersions` serializer method in the [application serializer unit tests](https://github.com/hashicorp/consul/blob/5fb9df1640cfa1aaf96228da1f44b31ecd6b825a/ui/packages/consul-ui/tests/unit/serializers/application-test.js#L10) if possible.

It would also be awesome to update our [mocks](https://github.com/hashicorp/consul/blob/2f20c77e4df9d663ee5f06d811c23dd9ad14db88/ui/packages/consul-ui/mock-api/v1/internal/ui/nodes#L28-L29) so that if agentless is enabled and it is a synthetic node, the consul version is empty. This would help to more closely mimic the behaviour of the real API and would be super helpful for dev and testing :)

File: ui/packages/consul-ui/app/serializers/application.js (hunk 2 above)
Comment (vijayraghav-io) [reply ui/packages/consul-ui/app/serializers/application.js:238]: @natmegs Yes! , was extra cautious :) , will update.

File: ui/packages/consul-ui/app/serializers/application.js (hunk 1 above)
Comment (vijayraghav-io) [reply ui/packages/consul-ui/app/serializers/application.js]: done

**Comment by vijayraghav-io** [comment]
@natmegs @david-yu 
done the updates, also updated mock-api for synthetic-node cases. 
And added test for getDistinctConsulVersions serializer method

**Comment by david-yu** [comment]
Thanks, I've added a backport. If you're able to create a changelog in the .changelog folder that would be amazing.

**Comment by vijayraghav-io** [comment]
added change log.

**Review by WenInCode — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
Backport failed @WenInCode. Run: https://github.com/hashicorp/consul/actions/runs/5944665998

**Comment by david-yu** [comment]
Re-triggering backport.

**Pull request merged**